src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -126,24 +199,35 @@ private function completeDefinition($id, Definition $definition)
 
                 if (isset($this->types[$typeHint->name])) {
                     $value = new Reference($this->types[$typeHint->name]);
+                    $addMethodCall = true;
                 } else {
                     try {
                         $value = $this->createAutowiredDefinition($typeHint, $id);
+                        $addMethodCall = true;
                     } catch (RuntimeException $e) {
                         if ($parameter->allowsNull()) {
                             $value = null;
                         } elseif ($parameter->isDefaultValueAvailable()) {
                             $value = $parameter->getDefaultValue();
                         } else {
-                            throw $e;
+                            // The exception code is set to 1 if the exception must be thrown even if it's a setter
+                            if (1 === $e->getCode() || $isConstructor) {
```

Comment:
Done (but I find the use of a specific code less intrusive than using accessors, here we add methods to a generic exception for an implementation detail of the `AutowirePass`).